doc/modules/calibration.rst

```diff
@@ -74,10 +74,12 @@ by showing the number of samples in each predicted probability bin.
 
 .. currentmodule:: sklearn.linear_model
 
-:class:`LogisticRegression` returns well calibrated predictions by default as it has a
+:class:`LogisticRegression` is more likely to return well calibrated predictions by default as it has a
 canonical link function for its loss, i.e. the logit-link for the :ref:`log_loss`.
-This leads to the so-called **balance property**, see [8]_ and
-:ref:`Logistic_regression`.
+This leads to the so-called **balance property**, see [8]_ and :ref:`Logistic_regression`.
+In the above example, the data is generated according to a linear mechanism, which is
+consistent with the :class:`LogisticRegression` model (the model is 'well specified'),
+which then returns accurate predictions from its `predict_proba` method.
```

Comment:
```suggestion
This leads to the so-called **balance property**, see [8]_ and :ref:`Logistic_regression`.
In the above example, the data is generated according to a linear mechanism, which is
consistent with the :class:`LogisticRegression` model (the model is 'well specified')
and the arbitrarily chosen value of the regularization parameter `C` happens to be
appropriate (neither too strong nor too low). As a consequence, this model returns
accurate predictions from its `predict_proba` method.
```

Reply:
As suggested above, I would mention the penalty with the balance property.

`C` is not "just right". It is well possible that a smaller value would lead to a better calibration.

Reply:
I've adapted the suggestions from the two of you (couldn't do it by accepting the suggestions on github because various suggestions were in conflict)